sympy/printing/preview.py

```diff
@@ -183,8 +183,12 @@ def preview(expr, output='png', viewer=None, euler=True, packages=(),
     try:
         workdir = tempfile.mkdtemp()
 
-        with open(join(workdir, 'texput.tex'), 'w', encoding='utf-8') as fh:
-            fh.write(latex_main % latex_string)
+        try:
+            with open(join(workdir, 'texput.tex'), 'w', encoding='utf-8') as fh:
+                fh.write(latex_main % latex_string)
+        except TypeError:  # Python 2's open() does not support encoding
```

Comment:
I think it's better to do it as you had it before. `import io` at the top and use `io.open` here.